port/main.c

```diff
@@ -33,38 +34,68 @@
 #include "filter.h"
 
 
+#define DEV_ROUTE_ID       0
+#define DEV_IFSTATUS_ID    1
+#define DEV_PF_ID          2
+#define DEV_LINKMONITOR_ID 3
+
+
 #define SNPRINTF_APPEND(fmt, ...) do { \
-		write = snprintf(buffer, size, fmt, ##__VA_ARGS__); \
-		if (offset > write) { \
-			offset -= write; \
-			write = 0; \
-		} else if (write > size) { \
-			return -ERANGE; \
-		} else if (offset) { \
-			write -= offset; \
-			memmove(buffer, buffer + offset, write); \
-			offset = 0; \
-		} \
-		size -= write; \
-		buffer += write; \
+		n = snprintf(buf, size, fmt, ##__VA_ARGS__); \
+		if (n >= size) \
+			goto overflow; \
+		size -= n; \
+		buf += n; \
 	} while (0)
 
 
-#ifdef LWIP_LINKMONITOR_DEV
 static struct {
-	int disconnected;
-	int reconnected;
-	int last_link;
-} main_common;
+	struct {
+		int busy;
+		char buf[1024];
```

Comment:
This is kindof wasteful on small tagets (`stm32l1`). Maybe we can avoid it somehow (either by ifdef or some other mechanism?)
@anglov, do You use these files at all? Maybe we can put them under single ifdef? (`LWIP_DEV_FILES` or something like that?)

Reply:
Thanks for metioning me here:
- I think it's ok for now to slighthly increase memory usage (it shouldn't break anything for now)
- Whole /dev files increase binary size and memory usage. So I would like to put those under ifdefs (or single ifdef as you suggest) as it not needed for small targets that don't have to have routing information, packet filter and even netstatus via /dev interface
- I'm not sure if putting it under one define is ok. Some configuration and status driver will be needed for PPP. It could be done by separate /dev interface. It have to be discussed how to achieve this.